evalml/tests/automl_tests/test_evalml_algorithm.py

```diff
@@ -0,0 +1,169 @@
+    X, y = X_y_binary
+    problem_type = "binary"
+    sampler_name = "Undersampler"
+
+    algo = EvalMLAlgorithm(X, y, problem_type, sampler_name)
+
+    assert algo.problem_type == problem_type
+    assert algo._sampler_name == sampler_name
+    assert algo.pipeline_number == 0
+    assert algo.batch_number == 0
+    assert algo.allowed_pipelines == []
+
+
+def test_evalml_algorithm_custom_hyperparameters_error(X_y_binary):
+    X, y = X_y_binary
+    problem_type = "binary"
+    sampler_name = "Undersampler"
+
+    custom_hyperparameters = [
+        {"Imputer": {"numeric_impute_strategy": ["median"]}},
+        {"One Hot Encoder": {"value1": ["value2"]}},
+    ]
+
+    with pytest.raises(
+        ValueError, match="If custom_hyperparameters provided, must be of type dict"
+    ):
+        EvalMLAlgorithm(
+            X,
+            y,
+            problem_type,
+            sampler_name,
+            custom_hyperparameters=custom_hyperparameters,
+        )
+
+    with pytest.raises(
+        ValueError, match="Custom hyperparameters should only contain skopt"
+    ):
+        EvalMLAlgorithm(
+            X,
+            y,
+            problem_type,
+            sampler_name,
+            random_seed=0,
+            custom_hyperparameters={"Imputer": {"impute_strategy": (1, 2)}},
+        )
+
+    with pytest.raises(
+        ValueError, match="Pipeline parameters should not contain skopt.Space variables"
+    ):
+        EvalMLAlgorithm(
+            X,
+            y,
+            problem_type,
+            sampler_name,
+            random_seed=0,
+            pipeline_params={"Imputer": {"impute_strategy": Categorical([1, 3, 4])}},
+        )
+
+
+def add_result(algo, batch):
+    scores = np.arange(0, len(batch))
+    for score, pipeline in zip(scores, batch):
+        algo.add_result(score, pipeline, {"id": algo.pipeline_number})
+
+
+@patch("evalml.pipelines.components.FeatureSelector.get_names")
+@pytest.mark.parametrize(
+    "automl_type",
+    [ProblemTypes.BINARY, ProblemTypes.MULTICLASS, ProblemTypes.REGRESSION],
+)
+def test_evalml_algorithm(
+    mock_get_names,
+    automl_type,
+    X_y_binary,
+    X_y_multi,
+    X_y_regression,
+):
+
+    if automl_type == ProblemTypes.BINARY:
+        X, y = X_y_binary
+        fs = "RF Classifier Select From Model"
+    elif automl_type == ProblemTypes.MULTICLASS:
+        X, y = X_y_multi
+        fs = "RF Classifier Select From Model"
+    elif automl_type == ProblemTypes.REGRESSION:
+        X, y = X_y_regression
+        fs = "RF Regressor Select From Model"
+    mock_get_names.return_value = ["0", "1", "2"]
+    problem_type = automl_type
+    sampler_name = None
+    algo = EvalMLAlgorithm(X, y, problem_type, sampler_name)
+    naive_model_families = [ModelFamily.LINEAR_MODEL, ModelFamily.RANDOM_FOREST]
+
+    first_batch = algo.next_batch()
+    assert len(first_batch) == 2
+    for pipeline in first_batch:
+        assert pipeline.model_family in naive_model_families
```

Comment:
I don't think this is blocking, but it's possible naive_model_families has multiple different families but each batch only has one of those types.  Might need to do a set equality in addition to the length equality to be sure.